### PR TITLE
Sample widgets update for v1.4

### PR DESCRIPTION
**Pull request by gavinr**
- required/breaking changes updates
- changed `d3` widget to functional-style component
- code formatting
- updated `version` and `exbVersion` to new version number

### CODE REVIEW

**Review by dmartine — APPROVED** [review]
Looks good

File: samples/widgets/d3/src/runtime/widget.tsx [hunk 1]
```diff
@@ -18,80 +18,78 @@
   A copy of the license is available in the repository's
   LICENSE file.
 */
-import { React, AllWidgetProps, jsx } from "jimu-core";
+import { jsx } from "jimu-core";
+import { useEffect, useRef } from "react";
```
Comment (qlqllu) [inline samples/widgets/d3/src/runtime/widget.tsx:22]: We should import react from `jimu-core`:
```
import {React} from 'jimu-core'
const {useEffect, useRef} = React
```

File: samples/widgets/feature-layer-class/src/runtime/widget.tsx [hunk 2]
```diff
@@ -19,15 +19,16 @@
 */
 import { React, IMDataSourceInfo, DataSource, DataSourceManager, DataSourceStatus, FeatureLayerQueryParams, AllWidgetProps, DataSourceComponent } from 'jimu-core';
 
-interface State {
+interface IState {
```
Comment (qlqllu) [inline samples/widgets/feature-layer-class/src/runtime/widget.tsx:22]: Add an `I` for interface is OK, but we don't add it in other widgets.

Comment (gavinr) [reply samples/widgets/feature-layer-class/src/runtime/widget.tsx:22]: ok, I will remove it for consistency.

Comment (gavinr) [reply samples/widgets/feature-layer-class/src/runtime/widget.tsx]: removed.

File: samples/widgets/d3/src/runtime/widget.tsx (hunk 1 above)
Comment (gavinr) [reply samples/widgets/d3/src/runtime/widget.tsx]: Thanks - updated.

**Pull request merged**